pol-core/pol/testing/testmisc.cpp

```diff
@@ -225,6 +225,38 @@ void test_convertquotedstring()
   test_dqs( "\" \\\"hi\"", " \"hi" );
 }
 
+void test_sanitizeUnicodeWithIso()
+{
+  std::string input;
+  std::string output;
+  std::string expected;
+
+  input = "Some weird characters: \x87 \x95 \x80.";
+  output = std::string(input);
+  expected = "Some weird characters: ‡ • €.";
+  Clib::sanitizeUnicodeWithIso(&output);
+  if ( output != expected )
+  {
+    INFO_PRINT << "sanitizeUnicodeWithIso fails!\n\tinput: " << input << "\n\toutput: " << output << "\n\texpected: " << expected << "\n";
+    UnitTest::inc_failures();
+  }
+  else
+    UnitTest::inc_successes();
+
+  input = "Maybe someone just wants to say \xC3\xA4 in CP-1252.";
+  output = std::string(input);
+  expected = "Maybe someone just wants to say Ã¤ in CP-1252.";
+  Clib::sanitizeUnicodeWithIso(&output);
+  if ( output != expected )
+  {
+    INFO_PRINT << "sanitizeUnicodeWithIso fails!\n\tinput:    " << input << "\n\toutput:   " << output << "\n\texpected: " << expected << "\n";
+    UnitTest::inc_failures();
+  }
+  else
+    UnitTest::inc_successes();
```

Comment:
This is the test that demonstrates the problem with the current implementation of `sanitizeUnicodeWithIso()`. It is impossible for any CP-1252 text passing through this function to contain the character combination `Ã¤` (and many other combinations that are valid UTF-8 surrogates).